app/assets/javascripts/discourse/app/lib/screen-track.js

```diff
@@ -85,6 +89,99 @@ export default class {
     this._anonCallback = cb;
   }
 
+  consolidateTimings(timings, topicTime, topicId) {
+    this._consolidatedTimings = this._consolidatedTimings || [];
+
+    let found = this._consolidatedTimings.find((elem) => elem[2] === topicId);
+
+    if (found) {
+      this._consolidatedTimings.removeObject(found);
+
+      const oldTimings = found[0];
+      Object.keys(oldTimings).forEach((id) => {
+        if (timings[id]) {
+          oldTimings[id] += timings[id];
+        }
+      });
+      found[1] += topicTime;
+      found[0] = Object.assign({}, timings, found[0]);
+    }
+
+    found = found || [timings, topicTime, topicId];
+    this._consolidatedTimings.unshift(found);
+
+    return this._consolidatedTimings;
+  }
+
+  sendNextConsolidatedTiming() {
+    if (!this._consolidatedTimings || this._consolidatedTimings.length === 0) {
```

Comment:
If you initialized `this._consolidatedTimings` and `this._blockSendingToServerTill` in the constructor, not only would it be more performant due to the way javascript handles object shapes, but you wouldn't need to check if they exist every time.

Reply:
sure done!